:memo: updated documentation to include ports and an overview (#841)
it-at-m/refarch: docs/templates/develop.md

The configuration in the `application.yml` file (inside the `appswitcher-server` directory of the stack) can be modified to include additional tools required for your specific project setup.
:::

> ### Ports and locale urls

Reviewer: ```suggestion
### Local services and ports
```

> The following table shows which local development services are served on which port.

Reviewer: ```suggestion
The following table shows which local development service is served on which port.
```

> 

Reviewer: Maybe add an information card here, that the links of the services can be clicked to directly open the available web UIs in the browser (if they are running)

Author: I think a link - which looks like a link - does not need explanation 😉

Reviewer: Yeah but people might think the link goes to some official documentation otherwise.

> | Service                                          | Standard port |

Reviewer: ```suggestion
| Service                                          | Default port |
```

| ------------------------------------------------ | :-----------: |

> | [Frontend (Vite Dev)](http://localhost:8081)     |     8081      |

Reviewer: Maybe add "Vite Development Server"? On the other side, we want users to develop frontend always through the API Gateway, so we could argue to remove the link here.

| [Webcomponent (Vite Dev)](http://localhost:8082) |     8082      |
| [API Gateway](http://localhost:8083)             |     8083      |

> | AppSwitcher                                      |     8084      |

Reviewer: I'd move AppSwitcher to a higher port maybe because its not really relevant while developing. It's just part of the stack.

Author: This would need adjustment of the port inside of the compose-stack. As of now appswitcher runs on `8084`.

Reviewer: Ok fine for now.

> | EAI (spring Boot)                                |     8085      |

Reviewer: Is Spring Boot really relevant here? I would just write EAI and Backend

Author: Yes - good question. But then we should also remove `Vite Dev` or any variation as of above too.

| Backend (Spring Boot)                            |     8086      |
| [Keycloak](http://localhost:8100)                |     8100      |
| PostgreSQL                                       |     5432      |
| [pgAdmin](http://localhost:5050)                 |     5050      |

> 
> ::: info Information
> This configuration could be changed depending on your project. The most current configuration can be viewed from your `stack/docker-compose.yaml`.
> :::

Reviewer: I'd formulate it something like:

```suggestion
::: danger Important
Depending on the project-specific configuration in `stack/docker-compose.yml` different ports might be used. This table only shows the default ports that ship with the `refarch-templates`.
:::
```

Author: Was this written by DeepL? 😄

Author: I would suggest to not use danger here - as these links are purely informational.
But i would be open to exchange `your` by `project-specific` - and a variation of the last sentence.

Reviewer: There would also be `warning` but I guess we haven't used it yet anywhere in the docs so let's stick with `information`. And no: This was written with my own english skills.

The configuration in the `application.yml` file (inside the `appswitcher-server` directory of the stack) can be modified to include additional tools required for your specific project setup.
:::

### Local services and ports

The following table shows which local development service is served on which port (services reachable inside the browser will have a direct `localhost` link).

| Service                               | Default port |
| ------------------------------------- | :----------: |

> | [Frontend](http://localhost:8081)     |     8081     |

Reviewer: Do you think we should keep the link here if we want developers to open Frontend through API Gateway?

Author: mhh good question - normally we dont want that. Same would be for the webcomponent, but this one is not included inside the gateway by default.

Author: Probably remove the link and keep it for the webcomponent

Author: ```suggestion
| [Frontend](http://localhost:8083)     |     8081     |
```

Author: Changed link to api-gw. This would be the normal workflow.

| Frontend (Vite Dev)     |          8081 | http://localhost:8081 |
| Webcomponent (Vite Dev) |          8082 | http://localhost:8082 |
| API Gateway             |          8083 | http://localhost:8083 |
| AppSwitcher             |          8084 | http://localhost:8084 |
| EAI (spring Boot)       |          8085 | http://localhost:8085 |
| Backend (Spring Boot)   |          8086 | http://localhost:8086 |
| Keycloak                |          8100 | http://keycloak:8100  |
| PostgreSQL              |          5432 | http://localhost:5432 |

> | pgAdmin                 |          5050 | http://localhost:5050 |

Author: This list includes only docker services found in another repository.

::: info Information
Depending on the project-specific configuration, differen tports might be used inside the `stack/docker-compose.yaml`.

> This table only shows default ports shipped inside the `refarch-templates`.

Reviewer: ```suggestion
Depending on the project-specific configuration, different ports might be used inside the `stack/docker-compose.yaml`.
This table only shows default ports shipped in the `refarch-templates`.
```

> This configuration could be changed depending on your project. The most current configuration can be viewed from your `docker-compose.yaml` inside the `stack` folder.

Author: resolved
